app/workers/travis_build_killer.rb

```diff
@@ -47,6 +45,12 @@ def kill_builds
           b.cancel
         end
       end
+
+      long_running_jobs.each do |b|
+        next unless b.running?
```

Comment:
But the first thing you do on https://github.com/ManageIQ/miq_bot/pull/132/files#diff-d80c65f3a5fe96f44114c253eb74c918R35 is to only select the running ones.